### PR TITLE
Add .gitignore to dev-kit for projects (+ config for a custom part)

### PR DESCRIPTION
**Pull request by OskarStark**

### CODE REVIEW

File: project/.gitignore.twig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/build
```
Comment (core23) [inline project/.gitignore.twig]:
```suggestion
/build/
/vendor/
```

**Comment by OskarStark** [comment]
Hmm, this would be the diff against the admin bundle:
```diff
diff --git a/.gitignore b/.gitignore
index cfcb6f20..6a9969b4 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,64 +1,7 @@
-/build
+/build/
 /vendor
-/docs/_build/*
 .php_cs.cache
 composer.lock
 phpunit.xml
-
-# clean up some non bower ready package
-Resources/public/vendor/admin-lte/bootstrap
-Resources/public/vendor/admin-lte/build
-Resources/public/vendor/admin-lte/dist/img
-!Resources/public/vendor/admin-lte/dist/img/boxed-bg.jpg # Only image called in AdminLTe.css
-Resources/public/vendor/admin-lte/dist/js/app.js
-Resources/public/vendor/admin-lte/dist/js/demo.js
-Resources/public/vendor/admin-lte/dist/js/pages
-Resources/public/vendor/admin-lte/documentation
-Resources/public/vendor/admin-lte/pages
-Resources/public/vendor/admin-lte/plugins
-Resources/public/vendor/admin-lte/*.html
-Resources/public/vendor/admin-lte/Gruntfile.js
-
-Resources/public/vendor/iCheck/skins/all.css
-Resources/public/vendor/iCheck/skins/*/*
-!Resources/public/vendor/iCheck/skins/square/blue.css
-!Resources/public/vendor/iCheck/skins/square/blue.png
-!Resources/public/vendor/iCheck/skins/square/blue@2x.png
-Resources/public/vendor/iCheck/icheck.js
-
-Resources/public/vendor/slimScroll/examples
-Resources/public/vendor/slimScroll/jquery.slimscroll.js
-
-Resources/public/vendor/bootstrap/grunt
-Resources/public/vendor/bootstrap/less
-Resources/public/vendor/bootstrap/test-infra
-Resources/public/vendor/bootstrap/Gruntfile.js
-
-Resources/public/vendor/jqueryui/themes/black-tie
-Resources/public/vendor/jqueryui/themes/blitzer
-Resources/public/vendor/jqueryui/themes/cupertino
-Resources/public/vendor/jqueryui/themes/dark-hive
-Resources/public/vendor/jqueryui/themes/dot-luv
-Resources/public/vendor/jqueryui/themes/eggplant
-Resources/public/vendor/jqueryui/themes/excite-bike
-Resources/public/vendor/jqueryui/themes/hot-sneaks
-Resources/public/vendor/jqueryui/themes/humanity
-Resources/public/vendor/jqueryui/themes/le-frog
-Resources/public/vendor/jqueryui/themes/mint-choc
-Resources/public/vendor/jqueryui/themes/overcast
-Resources/public/vendor/jqueryui/themes/pepper-grinder
-Resources/public/vendor/jqueryui/themes/redmond
-Resources/public/vendor/jqueryui/themes/smoothness
-Resources/public/vendor/jqueryui/themes/south-street
-Resources/public/vendor/jqueryui/themes/start
-Resources/public/vendor/jqueryui/themes/sunny
-Resources/public/vendor/jqueryui/themes/swanky-purse
-Resources/public/vendor/jqueryui/themes/trontastic
-Resources/public/vendor/jqueryui/themes/ui-darkness
-Resources/public/vendor/jqueryui/themes/ui-lightness
-Resources/public/vendor/jqueryui/themes/vader
-
-Resources/public/vendor/jquery/src
-
-Resources/public/vendor/jquery.scrollTo/demo
-Resources/public/vendor/jquery.scrollTo/tests
+/.phpunit.result.cache
+/docs/_build/
```

Shall we make additional paths configurable in the projects.yaml ?

**Comment by OskarStark** [comment]
@core23 shall we also exclude `phpstan.neon` ?

File: project/.gitignore.twig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/build/
+/vendor
```
Comment (core23) [inline project/.gitignore.twig]:
```suggestion
/vendor/
```

Comment (core23) [reply project/.gitignore.twig]: This should just ignore the root vendor dir

**Comment by OskarStark** [comment]
Diff updated

**Comment by OskarStark** [comment]
You approve? Shall we remove this completely from the admin bundle gitignore? Shall we make custom excludes per repo configurable in `projects.yaml`?

This PR is not ready yet

**Comment by core23** [comment]
> You approve? Shall we remove this completely from the admin bundle gitignore? Shall we make custom excludes per repo configurable in `projects.yaml`?
> 
> This PR is not ready yet

Sorry. I thought the last changes did fix this problem

**Comment by OskarStark** [comment]
@greg0ire @core23 I added a custom config part.

@greg0ire could you point me via a bash script like you already did to the repos which were different than the most? I would then apply their custom part to our project config

**Comment by greg0ire** [comment]
Here you go :) [gitignore.txt](https://github.com/sonata-project/dev-kit/files/4209710/gitignore.txt)

**Comment by OskarStark** [comment]
Thanks 🙏

File: config/projects.yml [hunk 3]
```diff
@@ -1,4 +1,63 @@
 admin-bundle:
+  custom_gitignore_part: |
+    # clean up some non bower ready package
+    Resources/public/vendor/admin-lte/bootstrap
```
Comment (core23) [inline config/projects.yml]: Do we need a full list. Can't we just uningnore `Resources/public/vendor/admin-lte/` (and the other root dirs)?

Comment (OskarStark) [reply config/projects.yml]: I am not sure WHY this was done, I just want to keep the current one. Improvments of the gitignore itself could be done in another PR, but I think when we merge https://github.com/sonata-project/SonataAdminBundle/pull/5461 it s obsolete anyway

**Comment by OskarStark** [comment]
God to go from my side, I will check all DevKit PRs once anyway

**Pull request merged**